common/src/main/java/com/scottlogic/deg/generator/fieldspecs/whitelist/DistributedSet.java

```diff
@@ -14,16 +14,16 @@
  * limitations under the License.
  */
 
-package com.scottlogic.deg.common.profile.constraints.atomic;
+package com.scottlogic.deg.generator.fieldspecs.whitelist;
 
-import com.scottlogic.deg.common.profile.Field;
 
 import java.util.Set;
 
-public class IsInNameSetConstraint extends IsInSetConstraint {
+public interface DistributedSet<T> {
```

Comment:
This would either create a tight coupling between the abstract class and the interface, or require exposing a method on the interface. We are trying to find a way to allow implementations to expose more information to other instances of the implementation, without exposing it publicly. As far as I'm aware this is impossible using purely Java's type system.